src/lib/components/DashCanvas.react.js

```diff
@@ -1,381 +1,119 @@
-	const show_undo = !(hide_buttons.includes("undo"));
-	const show_select = !(hide_buttons.includes("select"));
-	const show_rectangle = !(hide_buttons.includes("rectangle"));
-	var width_defined = this.props.width > 0;
-	var width = width_defined ? this.props.width : null;
-        return (
-	    <div className={this.props.className}>
-	    <SketchField name='sketch'
-		ref={(c) => this._sketch = c}
-                tool={toolsArray[this.props.tool.toLowerCase()]}
-                lineColor={this.props.lineColor}
-		width={width}
-		height={this.state.height}
-		forceValue={true}
-		backgroundColor='#ccddff'
-                lineWidth={this.props.lineWidth}/>
-	    {show_zoom &&
-	    <button style={styles.button}
-		    title="Zoom in"
-		    onClick={(e) => this._zoom()}>
-		<ZoomPlusIcon/>
-	    </button>
-	    }
-	    {show_zoom &&
-	    <button style={styles.button}
-		    title="Zoom out"
-		    onClick={(e) => this._unzoom()}>
-		<ZoomMinusIcon/>
-	    </button>
-	    }
-	    {show_pencil &&
-	    <button style={styles.button}
-		    title="Pencil tool"
-		    onClick={(e) => this._penciltool()}>
-		<EditIcon/>
-	    </button>
-	    }
-	    {show_line &&
-	    <button style={styles.button}
-		    title="Line tool"
-		    onClick={(e) => this._linetool()}>
-		<PlotLineIcon/>
-	    </button>
-	    }
-	    {show_rectangle &&
-	    <button style={styles.button}
-		    title="Rectangle tool"
-		    onClick={(e) => this._rectangletool()}>
-		<SquareIcon/>
-	    </button>
-	    }
-	    {show_pan &&
-	    <button style={styles.button}
-		    title="Pan"
-		    onClick={(e) => this._pantool()}>
-		<PanIcon/>
-	    </button>
-	    }
-	    {show_undo &&
-	    <button style={styles.button}
-		    title="Undo"
-		    onClick={(e) => this._undo()}>
-		<ArrowLeftIcon/>
-	    </button>
-		}
-		{show_undo &&
-	    <button style={styles.button}
-		    title="Redo"
-		    onClick={(e) => this._redo()}>
-		<ArrowRightIcon/>
-	    </button>
-	    }
-	    {show_select &&
-	    <button style={styles.button}
-		    title="Select"
-		    onClick={(e) => this._selecttool()}>
-		<TagOutlineIcon/>
-	    </button>
-	    }
-
-		<button style={styles.textbutton}
-		    title="Save"
-		    onClick={(e) => this._save()}>
-		    {this.props.goButtonTitle}
-	    </button>
-
-	    </div>
-
-        )
-     }
-
-
 }
 
-DashCanvas.defaultProps = {filename:'',
-			   json_data:'', image_content:'', trigger:0,
-			   width:500, height:500, scale:1, lineWidth:10,
-			   lineColor:'red', tool:"pencil", zoom:1,
-			   goButtonTitle:'Save', hide_buttons:[]};
+DashCanvas.defaultProps = {
+	filename: '',
+	json_data: '', image_content: '', trigger: 0,
+	width: 500, height: 500, scale: 1, lineWidth: 10,
+	lineColor: 'red', tool: "pencil", zoom: 1,
+	goButtonTitle: 'Save', hide_buttons: []
+};
 
 DashCanvas.propTypes = {
     /**
      * The ID used to identify this component in Dash callbacks
      */
-    id: PropTypes.string,
+	id: PropTypes.string,
```

Comment:
`\t`?

Reply:
weird..

Reply:
there seems to be a mishmash of whitespace characters in various files...